VeganVirus/VeganVirus/DesktopAction.cpp

```diff
@@ -0,0 +1,93 @@
+#include "DesktopAction.h"
+#include <cmath>
+
+#define BOUND_DIFF 10
+
+DesktopAction::DesktopAction(Draw* draw, MouseManager& mouseManager, DesktopManager& desktopManager)
+	:Action(DESKTOP_ACTION_PROGRESS, L"meat.ico"), _mouseManager(mouseManager), _desktopManager(desktopManager)
+{
+	_draw = draw;
+	_mouseManager = mouseManager;
+	_desktopManager = desktopManager;
+	_actionTime = 0;
+}
+
+void DesktopAction::start()
+{
+	_actionTime = DESKTOP_ACTION_TIME;
+}
+
+
+void DesktopAction::update(double dt)
+{
+	if (_actionTime < 0)
+		return;
+	_actionTime -= dt;
+	Point_t mouseTemp = _mouseManager.getMousePosition();
+	POINT mousePos = { mouseTemp.x, mouseTemp.y };
+	std::vector<POINT> iconSpeedArr = iconsMoveFromIcons();
+	// loping for each icon and seting is new postion on the screen base on interaction between the icons and the curser
+	for (int i = 0; i < _desktopManager.iconCount(); i++)
+	{
+		POINT iconPosition = _desktopManager.getIconPosition(i);
+		POINT iconMoveCurser = velocityBetweeenPoints(iconPosition, mousePos, CURSOR_INTERACTION_VELCITY_COEFFICIENT, 1);
+		POINT iconNewPosCurser = CheckIconPos({ (long)(iconPosition.x + (iconMoveCurser.x + iconSpeedArr[i].x) * dt), (long)(iconPosition.y + (iconMoveCurser.y + iconSpeedArr[i].y) * dt)});
+		_desktopManager.setIconPosition(i, iconNewPosCurser);
+	}
+}
+
+std::vector<POINT> DesktopAction::iconsMoveFromIcons()
```

Comment:
In my opinion, this method's name isn't descriptive of what the method does.
I suggest changing it to `computeIconInteractionVelocities`, `iconInteractionVelocities`, or something similar.